Platforms/QemuQ35Pkg/ConfigKnobs/ConfigDataGfx/ConfigDataGfx.c

```diff
@@ -0,0 +1,148 @@
+/** @file
+  Consumer module to locate conf data from variable storage, initialize
+  the GFX policy data and override the policy based on configuration values.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <Uefi.h>
+#include <PolicyDataStructGFX.h>
+#include <Protocol/Policy.h>
+
+#include <Ppi/ReadOnlyVariable2.h>
+#include <Ppi/Policy.h>
+#include <Library/DebugLib.h>
+#include <Library/PcdLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/PeiServicesLib.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PcdLib.h>
+#include <Library/PrintLib.h>
+#include <Library/MemoryAllocationLib.h>
+
+// XML autogen definitions
+#include <Generated/ConfigClientGenerated.h>
+
+// Statically define policy initialization for 2 GFX ports
+GFX_POLICY_DATA  DefaultQemuGfxPolicy[GFX_PORT_MAX_CNT] = {
+  {
+    .Power_State_Port = TRUE
+  },
+  {
+    .Power_State_Port = TRUE
+  }
+};
+
+/**
+  Helper function to translate GFX configuration data to GFX silicon policy.
+
+  @param[in]      PolicyInterface   Pointer to current policy protocol/PPI interface.
+  @param[in]      PlatformGfxPowerOnPort0 Value of PowerOnPort0 config knob.
+  @param[out]     GfxSiliconPolicy  Optional pointer to hold translated GFX silicon policy.
+                                    May be NULL with a zero PolicySize in order to determine
+                                    the size buffer needed.
+  @param[in, out] PolicySize        On input, this is the available size of GfxSiliconPolicy,
+                                    on output, this is the size of translated GfxSiliconPolicy.
+
+  @retval EFI_SUCCESS           The configuration is translated to policy successfully.
+  @retval EFI_INVALID_PARAMETER One or more of the required input pointers are NULL.
+  @retval EFI_BUFFER_TOO_SMALL  Supplied GfxSiliconPolicy is too small to fit in translated data.
+  @retval Others                Other errors occurred when getting GFX policy.
+**/
+STATIC
+EFI_STATUS
+ConvertGfxPolicyFromConfData (
+  IN      POLICY_PROTOCOL  *PolicyInterface,
+  IN      BOOLEAN          PlatformGfxPowerOnPort0,
+  OUT     GFX_POLICY_DATA  *GfxSiliconPolicy OPTIONAL,
+  IN OUT  UINT16           *PolicySize
+  )
+{
+  EFI_STATUS  Status;
+
+  if ((PolicyInterface == NULL) ||
+      (GfxSiliconPolicy == NULL) || (PolicySize == NULL))

```

Comment:
GfxSiliconPolicy is listed as an optional parameter. Is it optional?

Reply:
```suggestion
      (PolicySize == NULL) || ((GfxSiliconPolicy == NULL) && (*PolicySize != 0)))
```
I think the check probably should be more like above?